src/engine/Server.cpp

```diff
@@ -456,16 +458,25 @@ CPP_template_2(typename RequestT, typename ResponseT)(
                                 parsedOperation._originalString)));
     }
     if (parsedOperation.hasUpdateClause()) {
-      co_return co_await processUpdate(
-          std::move(parsedOperation), requestTimer, cancellationHandle, qec,
-          std::move(request), send, timeLimit.value());
+      AD_CORRECTNESS_CHECK(parsedOperation.hasUpdateClause());
+      PlannedQuery plannedQuery = co_await planQuery(
+          updateThreadPool_, std::move(parsedOperation), requestTimer,
+          timeLimit.value(), qec, cancellationHandle);
+      pq = plannedQuery;
+      co_return co_await processUpdate(std::move(plannedQuery), requestTimer,
+                                       cancellationHandle, std::move(request),
+                                       send);
     } else {
       AD_CORRECTNESS_CHECK(parsedOperation.hasSelectClause() ||
                            parsedOperation.hasAskClause() ||
                            parsedOperation.hasConstructClause());
-      co_return co_await processQuery(
-          parameters, std::move(parsedOperation), requestTimer,
-          cancellationHandle, qec, std::move(request), send, timeLimit.value());
+      PlannedQuery plannedQuery = co_await planQuery(
+          queryThreadPool_, std::move(parsedOperation), requestTimer,
+          timeLimit.value(), qec, cancellationHandle);
+      pq = plannedQuery;
```

Comment:
It seems like these two statements are the same regardless of the if branch. Can you move it in front of the if to reduce the duplication?

Reply:
If you apply my suggestion about avoiding `std::move` here, then you can just store the planned query in the optional directly and just pass a reference from optional and avoid the copy entirely.